crates/bevy_reflect/src/reflect.rs

```diff
@@ -313,6 +313,24 @@ where
         })
     }
 
+    /// For a type implementing [`PartialReflect`], combines `reflect_clone` and
+    /// `take` in a useful fashion, automatically constructing an appropriate
+    /// [`ReflectCloneError`] if the downcast fails.
+    ///
+    /// This is an associated function, rather than a method, because methods
+    /// with generic types prevent dyn-compatibility.
+    fn reflect_clone_and_take<T: 'static>(&self) -> Result<T, ReflectCloneError>
+    where
+        Self: TypePath + Sized,
```

Comment:
I'm a bit confused 🤔 , this looks like a method to me, and the `Self: Sized` bound should be enough to keep the trait dyn-compatible.

Reply:
Yeah "associated function" might be slightly confusing here. Personally, I think that part can be left out of the docs.

Reply:
Sorry, that comment reflects an earlier version of the code where this was an associated function. I will remove the sentence.